projects/go-lib/src/lib/components/go-off-canvas/go-off-canvas.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit, ViewChild, ComponentFactoryResolver } from '@angular/core';
+import { GoOffCanvasDirective } from './go-off-canvas.directive';
+import { GoOffCanvasService } from './go-off-canvas.service';
+import { GoOffCanvasItem } from './go-off-canvas.interface';
+
+import { fadeAnimation } from '../../animations/fade.animation';
+import { offCanvasAnimation } from '../../animations/off-canvas.animation';
+
+@Component({
+  selector: 'go-off-canvas',
+  templateUrl: './go-off-canvas.component.html',
+  styleUrls: [
+    './go-off-canvas.component.scss'
+  ],
+  animations: [
+    fadeAnimation,
+    offCanvasAnimation
+  ]
+})
+export class GoOffCanvasComponent implements OnInit {
+  currentOffCanvasItem: GoOffCanvasItem;
+  opened: boolean = false;
+
+  @ViewChild(GoOffCanvasDirective) goOffCanvasHost: GoOffCanvasDirective;
+
+  constructor(
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private goOffCanvasService: GoOffCanvasService
+  ){}
+
+  ngOnInit(): void {
+    this.goOffCanvasService.activeOffCanvasComponent.subscribe((goOffCanvasItem) => {
+      this.currentOffCanvasItem = goOffCanvasItem;
+      this.loadComponent();
+    });
+
+    this.goOffCanvasService.offCanvasOpen.subscribe((value) => {
+      this.opened = value;
+    });
+  }
+
+  loadComponent(): void {
```

Comment:
See above comment. I think we should type everything always even if Angular doesn't force us to. I think it will help us be more aware of the capabilities of typescript and the purpose our functions serve.